sortedm2m_tests/test_field.py

```diff
@@ -183,6 +183,16 @@ def get_ids(queryset):
             return [obj.id for obj in queryset]
         self.assertEqual(get_ids(shelf.books.all()), get_ids(books))
 
+    def test_base_class_str(self):
+        shelf = self.model.objects.create()
+        shelf.books.add(self.books[0])
+
+        through_model = shelf.books.through
+        instance = through_model.objects.first()
```

Comment:
unfortunately I forgot `first()` is not supported on old django versions.

Try rewriting this line as:

```python
instance = through_model.objects.all()[0]
```